app/src/main/java/org/fossasia/openevent/activities/TracksActivity.java

```diff
@@ -39,9 +46,11 @@ protected void onCreate(Bundle savedInstanceState) {
                 (CollapsingToolbarLayout) findViewById(R.id.collapsing_toolbar);
         collapsingToolbar.setTitle(track);
 
+
         RecyclerView recyclerView = (RecyclerView) findViewById(R.id.recyclerView);
         try {
-            sessionsAdapter = new SessionsAdapter(dbSingleton.getSessionbyTracksname(track));
+            List<Session> sessionList = dbSingleton.getSessionbyTracksname(track);
```

Comment:
Agreed, when data set gets large, this can happen. Let me read this and try to figure out what we can do.